examples/api/async_commit/async_commit.go

```
@@ -0,0 +1,89 @@
+*/
+func main() {
+	c, err := util.ReadConfig("../../util/config.yml")
+	if err != nil {
+		fmt.Printf(err.Error())
+	}
+
+	logger, err := logger.New(
+		&logger.Config{
+			Level:         "debug",
+			OutputPath:    []string{"stdout"},
+			ErrOutputPath: []string{"stderr"},
+			Encoding:      "console",
+			Name:          "bcdb-client",
+		},
+	)
+
+	conConf := &config.ConnectionConfig{
+		ReplicaSet: c.ConnectionConfig.ReplicaSet,
+		RootCAs:    c.ConnectionConfig.RootCAs,
+		Logger:     logger,
+	}
+
+	fmt.Println("Opening connection to database, configuration: ", c.ConnectionConfig)
+	db, err := bcdb.Create(conConf)
+	if err != nil {
+		fmt.Printf("Database connection creating failed, reason: %s\n", err.Error())
+		return
+	}
+
+	sessionConf := &config.SessionConfig{
+		UserConfig:   c.SessionConfig.UserConfig,
+		TxTimeout:    c.SessionConfig.TxTimeout,
+		QueryTimeout: c.SessionConfig.QueryTimeout}
+
+	fmt.Println("Opening session to database, configuration: ", c.SessionConfig)
+	session, err := db.Session(sessionConf)
+	if err != nil {
+		fmt.Printf("Database session creating failed, reason: %s\n", err.Error())
+		return
+	}
+
+	fmt.Println("Opening data transaction")
+	tx, err := session.DataTx()
+	if err != nil {
+		fmt.Printf("Data transaction creating failed, reason: %s\n", err.Error())
+		return
+	}
+
+	fmt.Println("Adding key, value: key1, val1 to the database")
+	err = tx.Put("bdb", "key1", []byte("val1"), nil)
+	if err != nil {
+		fmt.Printf("Adding new key to database failed, reason: %s\n", err.Error())
+		return
+	}
+
+	fmt.Println("Committing transaction")
+	txID, txReceipt, err := tx.Commit(false)
+	if err != nil {
+		fmt.Printf("Commit failed, reason: %s\n", err.Error())
+		return
+	}
+	l, err := session.Ledger()
+	if err != nil {
+		fmt.Printf(err.Error())
+		return
+	}
+	fmt.Println("Getting transaction receipt")
+	for txReceipt == nil {
+		time.Sleep(200 * time.Millisecond)
+		txReceipt, _ = l.GetTransactionReceipt(txID)
```

do not ignore the error